File: ironpythoninaction/chapter15/15.3/EmbeddingPlugin/PluginBase/Plugins.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;

namespace Plugins
{
    public class PluginBase
    {
        private string _name;
        public string Name
        {
            get { return _name; }
        }

        public PluginBase(string name)
        { _name = name; }

        virtual public void Execute(TextBox textbox)
        { }
    }
}

File: ironpythoninaction/chapter15/15.3/PluginEmbeddingVB/PluginBase/Plugins.vb
Imports System.Windows.Forms

Public Class PluginBase
    Private _name As String
    Public ReadOnly Property Name() As String
        Get
            Return _name
        End Get
    End Property

    Public Sub New(ByVal name As String)
        _name = name
    End Sub

    Public Overridable Sub Execute(ByVal textbox As TextBox)
    End Sub
End Class
